index.js

```diff
@@ -12,6 +12,27 @@ function isNegative(pattern) {
 	return pattern[0] === '!';
 }
 
+function isString(value) {
+	return typeof value === 'string';
+}
+
+function validatePatternsArray(patterns) {
+	if (isString(patterns)) {
+		// bail early, the rest will easier this way
+		return;
+	}
+
+	var err = new TypeError('patterns must be a string or an array of strings');
+
+	if (!Array.isArray(patterns)) {
+		return err;
+	}
+
+	if (patterns.filter(isString).length !== patterns.length) {
```

Comment:
`patterns.every(isString)`

You can also move both checks into one `if` statement.